File: spring-boot-websocket-raw/src/examples/javascript/websocket-client.js
class ExampleClient {

    // Setup outputs for information we receive from WebSocket server
    constructor() {
        this.status = document.createElement("input");
        this.status.value = "fetching ...";
        document.body.appendChild(this.status);
        this.die = document.createElement("input");
        this.die.value = "waiting ...";
        document.body.appendChild(this.die);
    }

    // Start WebSocket connection to server and set handlers for WebSocket events
    connect() {
        const self = this;
        const socket = new WebSocket("ws://localhost:8080/websocket/json/messages");
        socket.onopen = function () {
            self.onopen(socket);
        }
        socket.onmessage = function (event) {
            self.onmessage(socket, event);
        }
        socket.onclose = function () {
            console.log("WebSocket connection closed");
            self.status.value = "Closed";
        }
        socket.onerror = function (event) {
            console.log("WebSocket error " + event);
            self.status.value = "ERROR";
        }
    }

    onopen(socket) {
        console.log("WebSocket connection opened");
        this.status.value = "Connected"
        ExampleClient.requestDiceRoll(socket);
        // Better to use JSON.stringify(object) for safety, but strings will work
        socket.send("{ \"request\": 0 }");

        setInterval(function () {
            ExampleClient.requestDiceRoll(socket);
        }, 1000);
    }

    static requestDiceRoll(socket) {
        // Prefer JSON stringify so you don't have to even think about escaping.
        socket.send(JSON.stringify({"request": 1, "dice": 1, "sides": 10}));
    }

    // Processes messages received from WebSocket server (if/else on id does not
    // scale well if there are many message types - this is just a quick example)
    onmessage(socket, event) {
        const message = event.data; // Assuming the server sends plain text messages

        // Handle the incoming message
        console.log("Received message:", message);
        const parts = JSON.parse(message);
        const content = parts["content"];

        if (content) {
            const id = parts["id"];
            if (id === 200) {
                this.status.value = content["sessionCount"] + " client(s)";
            } else if (id === 1) {
                this.die.value = content["rolls"][0];
            }
        }
    }

}

File: spring-boot-websocket-raw/src/examples/python/websocket-client.py
#!/usr/bin/python3
"""

Python example implementation that connects to example spring-boot WebSocket service.

"""

import sys
import websocket


def on_message(ws, message):
    print(message, flush=True) # flush buffers on each print for example of piping to jq


def on_error(ws, error):
    print(error, file=sys.stderr)


def on_close(ws, close_status_code, close_msg):
    print(f'WebSocket closed ({close_status_code}: {close_msg}))', file=sys.stderr)
    sys.exit(0)


def on_open(ws):
    print('WebSocket opened, requesting info', file=sys.stderr)
    ws.send('{ "request": 0 }')
    ws.send('{ "request": 1, "dice": 5 }')


if __name__ == "__main__":
    websocket.enableTrace(False)  # Set to True for more diagnostics
    server = websocket.WebSocketApp("ws://localhost:8080/websocket/json/messages", on_open=on_open,
                                    on_message=on_message, on_error=on_error, on_close=on_close)
    server.run_forever(ping_interval=60, ping_timeout=10)
